src/thread/linked_list.cr

```diff
@@ -6,7 +6,7 @@ class Thread
   #
   # Thread-safe doubly linked list of `T` objects that must implement
   # `#previous : T?` and `#next : T?` methods.
-  struct LinkedList(T)
+  class LinkedList(T)
```

Comment:
What is the cause for this change?

Reply:
LinkedList has a `Mutex` which was previously immutable but now it's mutable (has `@lock_count`) so using a struct means that locking effectively doesn't work. This has to be a class. But @waj or @bcardiff can confirm this.

Reply:
The lazy initialization `Fiber.@@fibers` plus the `Fiber.fiber` getter didn't play nice if the linked list was a value type. `Fiber.fiber.push` would've been modifying copy instead of the expected list.

Reply:
This doen't need to, `Mutex` is a class, thus passed by reference, even if inside a struc.
```cr
class Thread::Mutex
  getter lock_count : Int32 = 0
  def inc_lock
    @lock_count += 1
  end
end

def some_method(obj)
  obj.@mutex.inc_lock
end

ll = Thread::LinkedList(Nil).new
puts "before call: #{ll.@mutex.lock_count}"  #=> 0
some_method ll
puts "after call: #{ll.@mutex.lock_count}"   #=> 1
```

Reply:
This isn't needed, `Mutex` is a class, thus passed by reference, even if inside a struct.
```cr
class Thread::Mutex
  getter lock_count : Int32 = 0
  def inc_lock
    @lock_count += 1
  end
end

def some_method(obj)
  obj.@mutex.inc_lock
end

ll = Thread::LinkedList(Nil).new
puts "before call: #{ll.@mutex.lock_count}"  #=> 0
some_method ll
puts "after call: #{ll.@mutex.lock_count}"   #=> 1
```

Reply:
What I said before applies to the whole list. It has nothing to do with the mutex IIRC

Reply:
@bcardiff sorry I was responding to @asterite , I didn't see your message.

Reply:
Yeah, I don't know what I was thinking when I replied that.